backend/src/src-licenseinfo/src/main/java/org/eclipse/sw360/licenseinfo/parsers/SPDXParserTools.java

```diff
@@ -117,6 +122,20 @@ private static String getLocalName(String label) {
         for (int i = 0; i < label.length(); i++) {
             // label contain invalid NCName
             if (!XMLChar.isNCName(label.charAt(i))) {
+                if (Util.splitNamespaceXML(label) == label.length()) {
+                    try {
+                        label = URLDecoder.decode(label, StandardCharsets.UTF_8.name());
+                        URL url = new URL(label);
+                        if (null != url.getRef()) {
+                            return url.getRef();
+                        } else {
+                            String path = url.getPath();
```

Comment:
We can do a null check for the path variable as  well similar we did for the Ref variable, just in case to avoid any null pointer.

Reply:
For a valid URL, path cannot be null (can be empty string), So we don't need Null check here unlike ref which can be null.

Reply:
Thanks for the clarification.